packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.spec.ts

```diff
@@ -167,4 +167,71 @@ describe("XYChartComponent", () => {
             expect(spy).toHaveBeenCalledWith({ payload: timeframe });
         });
     });
+
+    describe("isSeriesInteractive", () => {
+        let initializationChanges: SimpleChanges;
+
+        beforeEach(() => {
+            initializationChanges = {
+                widgetData: { isFirstChange: () => false } as SimpleChange,
+                configuration: { isFirstChange: () => false, currentValue: {} } as SimpleChange,
+            };
+        });
+
+        it("should return true if the series is set to interactive", () => {
+            component.configuration = { interaction: "series" } as ITimeseriesWidgetConfig;
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
+
+        it("should return true if the dataSeries has 'link' populated", () => {
+            component.widgetData = { series: [{} as ITimeseriesWidgetData] };
+            component.widgetData.series[0] = {
+                id: "id",
+                name: "name",
+                description: "description",
+                link: "www.link.com",
+                data: {} as ITimeseriesWidgetSeriesData[],
+            }
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
+
+        it("should return true if the dataSeries has 'secondaryLink' populated", () => {
+            component.widgetData = { series: [{} as ITimeseriesWidgetData] };
+            component.widgetData.series[0] = {
+                id: "id",
+                name: "name",
+                description: "description",
+                secondaryLink: "www.link.com",
+                data: {} as ITimeseriesWidgetSeriesData[],
+            }
+            component.ngOnChanges(initializationChanges);
+            expect(component.isSeriesInteractive(component.chartAssist.legendSeriesSet[0])).toBe(true);
+        });
+
+        it("should communicate interaction on the EventBus when series is interactive", () => {
+            const spy = spyOn(eventBus.getStream(INTERACTION), "next").and.callThrough();
+            component.configuration = { interaction: "series" } as ITimeseriesWidgetConfig;
+            component.ngOnChanges(initializationChanges);
+            component.onPrimaryDescClick(component.chartAssist.legendSeriesSet[0]);
+            const interactionData = {
+                payload: { data: component.chartAssist.legendSeriesSet[0], interactionType: TimeseriesInteractionType.Series },
+                id: "INTERACTION",
+            };
+            expect(spy).toHaveBeenCalledWith(interactionData);
+        });
```

Comment:
This test seems out of place in the isSeriesInteractive describe block. Maybe it deserves an onPrimaryDescClick describe block of its own?

Reply:
And, maybe add a test for the following description as well:
```
it("should not communicate interaction on the EventBus when series is not interactive", () => {
```